tests/compile_upp.sh

```diff
@@ -60,7 +64,17 @@ if [[ $MACHINE_ID != "unknown" ]]; then
       module purge
    fi
    module use $PATHTR/modulefiles
-   modulefile=${MACHINE_ID}
+   if [[ $compiler == "intel" ]]; then
+      modulefile=${MACHINE_ID}
+   else
+      modulefile=${MACHINE_ID}_${compiler}
+   fi
+   if [ -f "${PATHTR}/modulefiles/${modulefile}" ]; then
```

Comment:
@mkavulich On WCOSS2, we use modulefile in lua format. You might add changes as
if [ -f "${PATHTR}/modulefiles/${modulefile}" -o \
     -f "${PATHTR}/modulefiles/${modulefile}.lua" ]; then

Reply:
@WenMeng-NOAA Good catch, I have pushed this change. I have tested the build and running on Cheyenne (intel and gnu), and the build on Hera, Jet, and Orion.  I will need someone else to test other platforms as I do not have access.